encode_test.go

```diff
@@ -297,68 +822,107 @@ func TestEncoder_StatusCodeWithMessage(t *testing.T) {
 			t.Parallel()
 			recorder := httptest.NewRecorder()
 			encoder := NewEncoder(recorder)
-			if err := tt.statusWithMessageFunc(encoder, tt.args.message, tt.args.err); (err != nil) != tt.wantErr {
-				t.Errorf("Encoder.StatusCodeWithMessage() error = %v, wantErr %v", err, tt.wantErr)
+			err := tt.encodeMethod(encoder, tt.args.message, tt.args.a, tt.args.err)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Encoder.Method() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if CauseIsError(err) != tt.wantContainsError {
+				t.Errorf("ContainsError() = %v, wantContainsError %v", err, tt.wantContainsError)
 			}
 
 			if recorder.Result().StatusCode != tt.wantStatus {
 				t.Errorf("Decoder.DecodeRequest() wanted status code %d, got %d", http.StatusUnauthorized, recorder.Result().StatusCode)
 			}
 
+			if tt.wantMessage == "" {
+				return
+			}
+
 			var bod MessageResponse
 			if err := json.NewDecoder(recorder.Result().Body).Decode(&bod); err != nil {
 				t.Fatal("failed to decode body")
 			}
 
-			if bod.Message != tt.args.message {
-				t.Errorf("Encoder.StatusCodeWithMessage() want message = %s, got = %s", tt.args.message, bod.Message)
+			if bod.Message != tt.wantMessage {
+				t.Errorf("Encoder.StatusCodeWithMessage() want message = %s, got = %s", tt.wantMessage, bod.Message)
 			}
 		})
 	}
 }
 
-func TestEncoder_Ok(t *testing.T) {
+func TestEncoder_ClientMessage(t *testing.T) {
 	t.Parallel()
 
-	ctrl := gomock.NewController(t)
-	msg := "Testing"
 	type args struct {
 		message string
 		err     error
 	}
 	tests := []struct {
-		name         string
-		e            *Encoder
-		args         args
-		wantErr      bool
-		wantStatus   int
-		setupEncoder func(w http.ResponseWriter) HTTPEncoder
+		name       string
+		e          *Encoder
+		args       args
+		wantStatus int
```

Comment:
'message' should be replaced by a 'wantMessage' field that is NOT part of the args struct. The message isn't used as an argument.

Reply:
Done